src/dispatch/BasicAuthentication.ts

```diff
@@ -27,21 +27,27 @@ export class BasicAuthentication extends Authentication {
      */
     protected AnonymousCognitoCredentialsProvider =
         async (): Promise<AwsCredentialIdentity> => {
-            return this.cognitoIdentityClient
-                .getId({
-                    IdentityPoolId: this.config.identityPoolId as string
-                })
-                .then((getIdResponse) =>
-                    this.cognitoIdentityClient.getOpenIdToken(getIdResponse)
-                )
-                .then((getOpenIdTokenResponse) =>
-                    this.stsClient.assumeRoleWithWebIdentity({
-                        RoleArn: this.config.guestRoleArn as string,
-                        RoleSessionName: 'cwr',
-                        WebIdentityToken: getOpenIdTokenResponse.Token
-                    })
-                )
-                .then((credentials: AwsCredentialIdentity) => {
+            let retries = 1;
```

Comment:
question: why do we allow only one retry?

Reply:
The goal of the retry is to address the case when the identity Id is invalid. In this case, we only need on re-try after the bad identity Id has been removed from browser storage.

At a higher level, we want to minimize the impact of the RUM instrumentation on the performance of the web application. If the situation is not recoverable, then we don't want to keep re-trying.

Reply:
Retries should also be extended to PutRumEvents. A future PR can consolidate the retry logic, and maybe add configuration for it.